### ISSUE TITLE
[NoDrop 1.1.0.15] NullReferenceException since the last update (VANILLA) 455509

### ISSUE THREAD

**Issue by michaelvolz** [O-n-y/oxy.mods#45]
supportedContent: VANILLA_ID
lastWorkingBuild: 455509

```
NullReferenceException

UnityEngine.GameObject.GetComponent[T] () (at <b8388ffefd6046f8a7b4080349da72eb>:0)
Ony.OxygenNotIncluded.NoDrop.CarriedItemsComponent.FixedUpdate () (at <bac14f8a09924b34a319e61f4e76b512>:0)

```
Happens after some time, only sporadically.

Great Mod, keep up the good work please!

**Comment by O-n-y** [comment]
Make a proper issue request,  provide mod version you have installed.

**Comment by O-n-y** [comment]
Re-open, once provided more details

**Comment by O-n-y** [comment]
There is still no information about which exact version of the mod you have. 
You can check it in the file itself or using Mod Manager.

**Comment by O-n-y** [comment]
Please do read carefully how to report issue:
HowTo: https://github.com/O-n-y/oxy.mods

**Comment by jpetays** [comment]
How to report bug:
**Go to: https://github.com/O-n-y/oxy.mods/issues**
-> I am following instructions provided on the above mentioned page best I can

Create new issue with mod label
**Write full description of issue and the way to reproduce it**
-> the issue is that Ony.OxygenNotIncluded.NoDrop.CarriedItemsComponent throws NRE unexpectedly while playing the game.
For me this has happened twice since I installed the mod on 3 Jan @ 7:19pm. AFAIK both happened during last month or so.

Provide mod version you have (you can see it by using my mod "Mod Manager" https://steamcommunity.com/sharedfiles/filedetails/?id=1843965353)
-> Screenshot attached, need more instructions to provide more details if this is not what you want.

Attach game save file (game should be started with any other mod enabled, to be sure issue not caused by some other mod)
-> Two save files attached, one after when the NRE has happened and one after it has not happened yet (and some other mods unused removed).

If I could find the file for the code (either .cs or binary MSIL) I could see it for myself to try to find all components in FixedUpdate() that are accessed without checking if they can be null.

![mod_manager_issue_45](https://user-images.githubusercontent.com/44498878/116763259-dd67a080-aa25-11eb-988b-226a9b1a201b.png)


[Cesspool.zip](https://github.com/O-n-y/oxy.mods/files/6408502/Cesspool.zip)

**Comment by O-n-y** [comment]
Your screenshot not done with Mod Manager (this is some other mod), please provide mod version you are using.

**Comment by jpetays** [comment]
Now I got the correct mod manager - this is great!

Release_DLC1.Mod.NoDrop.dll version is 1.1.0.15 (on my mod folder).

I checked out CarriedItemsComponent FixedUpdate() and it seems that GetComponent is called three times in logging statements using **input.GetComponent**.
If I understand it correctly, input is checked against NULL in foreach (GameObject input in this.GetItems()) but not in for loop that comes later: for (int index = 0; index < items.Count; ++index).
Of course DropManager.Drop can return false if input is NULL so last logging statement is guarded against null as well?

The call stack is so short that I assume that NRE comes in FixedUpdate() itself but that may not be the case?

![oni_mods_by_oby](https://user-images.githubusercontent.com/44498878/116781630-c9a35500-aa8c-11eb-909e-3f7d36a2edcc.gif)

**Comment by alishahb** [comment]
thank you for the provided information, will be checking it out

**Comment by O-n-y** [comment]
thank you for the provided information will be checking it out!

**Comment by O-n-y** [comment]
Running your save file, not yet able to reproduce, how often do this happen?
Could you maybe provide save file, when it should happen in a bit? (i.e. game was saves and then it happen)
So it would be more easy to reproduce.

**Comment by O-n-y** [comment]
I managed to reproduce, fix should be in next release build

**Comment by O-n-y** [comment]
Fixed in  1.1.1.16
